### DISCUSSION TITLE
Once this hack is flashed, do i still need the wyze app?

### DISCUSSION THREAD

**Discussion by balajeek** [gtxaspec/wz_mini_hacks#81] [General]
All i wanted is the RTSP stream, for now i have 2 wyze v2 cams and using with the wyze rtsp firmware.  Just got the wyze cam v3 floodlight cam and flashed with this wz-mini-hack yesterday.  Do i still need the wyze app ?  I am trying to keep everything in LAN and I use frigate to handle my camera recording.

Thanks.

**Comment by gtxaspec** [comment]
you need the app to set everything up initially, but beyond that, not really

**Comment by virmaior** [comment]
> All i wanted is the RTSP stream, for now i have 2 wyze v2 cams and using with the wyze rtsp firmware.

doesn't the wyze rtsp firmware need to connect through the app somehow to work?

**Comment by gtxaspec** [comment]
you need to initially setup the rtsp username and password using the app if i recall correctly.

**Comment by Bob-AOL** [comment]
So the Wyze RTSP beta firmware is a pre-requisite for the mini-hacks RTSP implementation to work? That makes no sense. If that's the case, why not just use the Wyze firmware instead?

That being said, the hack RTSP implementation indeed does not work in the flashed V3 I just fired up which still has the current production firmware installed -- 4.36.9.139, fwiw. 

I can't see two RTSP implementations being required just to make one function...., especially since the hacks are purported to work with any version of factory firmware on any of the listed devices.

Seems like this caveat would be fairly well documented if it is an actual requirement for the mini-hack RTSP function to work..

**Reply by gtxaspec** [reply]
No, the rtsp beta firmware has nothing to do with wz_mini at all, and is not required.

**Reply by Bob-AOL** [reply]
Thanks for the swift response. Any clues to why I can't get RTSP functioning? Everything seems to have gone quite smoothly so far and the firmware seems to boot without a hitch according to the log file. Is the SSH authentication a pre-requisite for full functionality?

**Reply by gtxaspec** [reply]
you can edit the wz_mini.conf file on your pc, but the preferred method is ssh for configuration and debugging.    Is RTSP enabled in the wz_mini.conf file?

**Reply by Bob-AOL** [reply]
Enabled and configured with the same login credentials as my other V3s. Only the main stream is active at present. Same basic parameters as used for the other cams in my Blue Iris setup.

**Reply by gtxaspec** [reply]
check the logs folder, wz_user.log  You should see something regarding the rtsp server there

**Reply by Bob-AOL** [reply]
Will need to pull the sd card since no SSH access yet. I'm guessing it's something simple (depending on the granularity provided in the log).
I appreciate the help!

**Reply by Bob-AOL** [reply]
Not seeing any obvious complaints in the logs, but not exactly sure what I'd be looking for, either. RTSP params are straightforward. Ports 22, 80 , 554, and 4000 are shown as open.

RTSP_LOGIN="admin"
RTSP_PASSWORD="*******"
RTSP_PORT="554"

RTSP_HI_RES_ENABLED="true"
RTSP_HI_RES_ENABLE_AUDIO="false"
RTSP_HI_RES_FPS="20"
RTSP_HI_RES_MAX_BITRATE="2048"
RTSP_HI_RES_TARGET_BITRATE="1024"
RTSP_HI_RES_ENC_PARAMETER="2"

**Reply by Bob-AOL** [reply]
I usually see bitrate as kB/s but just followed the example in the readme and went with the above.

**Reply by gtxaspec** [reply]
if you only have hi_res enabled, the url would be `rtsp://x.x.x.x:554/unicast`

is that the url you are using? can you test with an app like VLC to check connectivity?

**Reply by Bob-AOL** [reply]
I'm trying to bring it up in Blue Iris via their vanilla config template using the basic parameters there -- ip address, RTSP on port 554 and the login credentials. Everything is there except the stream. There's also a WyzeCam v3 RTSP template that applies their default url which doesn't work either. I'll try your url suggestion in Blue Iris and VLC and see where it goes.

**Reply by Bob-AOL** [reply]
Bingo! Came right up with VLC, albeit with quite a bit of lag. Should be a whole different story with the Blue Iris rig. Now where did I put that ethernet to mini-usb adapter.....? LOL

Thanks so much for all your dedicated work and support. Very much appreciated!

**Reply by dcCMPY** [reply]
Hi @Bob-AOL 
Similar situation albeit im using synology surveillance .... I can get my RTSP stream to work via VLC, but not within the Synology app
did you get Blue Iris to work ?

**Reply by Bob-AOL** [reply]
Yes. See post above from gtxaspec for starters. Since there's essentially no sub-stream even available, it should be disabled in the .conf to achieve a default unicast path -- which I found unnecessary to actually specify for whatever reason(s) in my particular cases.

The setup in Blue Iris is camera type Wyze - model *V3 RTSP - RTSP port 554 - Onvif port 8999 - main stream path /live
rtsp;//user:userpass@x.x.x.x:554/live 

Default 1080 resolution at 22fps with Intel hardware acceleration. Nothing exotic or out of the ordinary.

Works flawlessly without lag for both wireless and wired ethernet adapted V3's with Blue Iris, VLC, and TinyCam. There's a degree of lag with TinyCam due to the lack of available hardware horsepower on an AndroidTV box, but is still watchable.

**Reply by Bob-AOL** [reply]
edit to the above for clarification: The BI camera configuration is actually an http:// URL, not RTSP:// as would otherwise be the case for a VLC stream.

**Comment by pjuhl2313** [comment]
For those using Ethernet for their Wyze cams, what Ethernet to micro-usb adapters are working for you?
